app/src/main/java/camp/nextstep/controller/LoginController.java

```diff
@@ -1,38 +1,59 @@
 package camp.nextstep.controller;
 
-import camp.nextstep.domain.User;
 import camp.nextstep.dao.InMemoryUserDao;
+import camp.nextstep.domain.User;
+import com.interface21.context.stereotype.Controller;
+import com.interface21.web.bind.annotation.RequestMapping;
+import com.interface21.web.bind.annotation.RequestMethod;
+import com.interface21.webmvc.servlet.ModelAndView;
 import jakarta.servlet.http.HttpServletRequest;
 import jakarta.servlet.http.HttpServletResponse;
-import com.interface21.webmvc.servlet.mvc.asis.Controller;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class LoginController implements Controller {
-
+@Controller
+public class LoginController {
     private static final Logger log = LoggerFactory.getLogger(LoginController.class);
 
-    @Override
-    public String execute(final HttpServletRequest req, final HttpServletResponse res) throws Exception {
+    @RequestMapping(value = "/login/view", method = RequestMethod.GET)
+    public ModelAndView show(final HttpServletRequest req, final HttpServletResponse res) {
+        return UserSession.getUserFrom(req.getSession())
+                          .map(user -> {
+                              log.info("logged in {}", user.getAccount());
+                              return ModelAndView.jspView("redirect:/index.jsp");
+                          })
+                          .orElse(ModelAndView.jspView("/login.jsp"));
```

Comment:
jsp는 spring에서 제공하는 기본 템플릿 엔징중 하나인데 스프링 부트로 넘어오면서 공식 지원은 하지 않는 템플릿 엔진이기도 합니다. 
즉 별도의 모듈추가를 해주지 않을경우 기본적으로 jsp를 지원하지 않죠. 

ModelAndView객체가 jspView라는 함수로 jsp라는 특정 확장자에 대응하도록 만든다면 해당 객체는 jsp라는 도메인에 의존하게 되는 문제가 발생하는데요. 저는 그래서 함수를 좀 더 범용적으로 만들고 실제로 처리하는건 ViewResolver를 통해서 하는게 어떨까 생각하네요 

```suggestion
                          .orElse(ModelAndView.from("/login.jsp"));
```

Reply:
스프링 코드를 참조해서 ModelAndView 의 생성자로 문자열을 넘기고, DispatcherServlet 과 viewResolvers 의 조합으로 처리하도록 수정해 보았습니다. 덕분에 관련 코드를 읽어볼 수 있어서 좋았던 것 같습니다.